_rules/link-skip-block-content-7b576d.md

```diff
@@ -0,0 +1,814 @@
+---
+id: 7b576d
+name: Link for skipping block of content
+rule_type: atomic
+description: |
+  This rule checks that blocks of content can be skipped by a link at their beginning
+accessibility_requirements:
+  wcag-technique:G123: # Adding a link at the beginning of a block of repeated content to go to the end of the block
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS styling
+  - DOM Tree
+  - Language
+acknowledgments:
+  authors:
+    - Christina Adams
+    - Jean-Yves Moyen
+  assets:
+    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
+    - Image from a Ming Dynasty edition of the Romance of the Three Kingdoms, original kept in the library holdings of Peking University.
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectation
+
+There exists a [semantic segmentation][] of the test target such that for each [block][] in this [segmentation] which is before the [main block of content][] and contains at least one [block of repeated content][], either the last [focusable][] element which is before any [focusable][] element inside this [block][], or the first [focusable][] element which is inside this [block][]:
+
+- has a [semantic role][] of `link`; and
+- is [included in the accessibility tree][]; and
+- is [visible][] when [focused][]; and
+- can be [activated][] by use of keyboard; and
+- has an [accessible name][] that communicates that it skips this [block][]; and
+- when [activated][], moves keyboard focus to a node which is [at the end][] of this [block][].
```

Comment:
G123 states something else:
>Check that activating the link moves the focus to the content **immediately** after the block.

Reply:
"at the end" is an attempt to have an objective definition of what "immediately after" means.
Do you think there is a problem with this definition?

Reply:
then why the failed example 10 failed as the focus moved to a node which is at the end of that block?

Reply:
In Failed example 10, the focus move to the last `p` element **inside** the block of repeated content.
This does not match the definition of "at the end".

On a non-technical side, in Failed Example 10, user is forced to read/hear part of the block of repeated content (the last `p`) and cannot fully skip it, which is why I think it fails 2.4.1.